docs/references/astro/ui-react.mdx

```diff
@@ -0,0 +1,127 @@
+---
+title: Use Clerk with Astro and React
+description: Learn how to user Clerk inside an Astro app with React
+---
+
+# Use Clerk with Astro and React
+
+Astro supports intregrations with many UI Frameworks. This guide we teach you how to untilize [`@clerk/astro`](/docs/references/astro/overview) and use [Clerk's components](https://clerk.com/docs/pr/1255/components/overview#ui-components), hooks and the new stores. If you have not set up you Astro applicatoin to work with Clerk, visit the [Astro quickstart.](https://clerk.com/docs/pr/1255/quickstarts/astro)
+
+
+<Steps>
+### Install `@astrojs/react`
+
+Alternatively can also read the [official Astro documentation](https://docs.astro.build/en/guides/integrations-guide/react/)
+
+<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+  ```bash filename="terminal"
+  npx astro add react
+  ```
+
+  ```bash filename="terminal"
+  yarn astro add react
+  ```
+
+  ```bash filename="terminal"
+  pnpm astro add react
+  ```
+</CodeBlockTabs>
+
+### Update `astro.config.mjs`
+
+Add the react integration inside your Astro configuration file.
+
+```ts filename="astro.config.mjs" {3,9}
+import { defineConfig } from "astro/config";
+import node from "@astrojs/node";
+import react from "@astrojs/react";
+import clerk from "@clerk/astro";
+
+export default defineConfig({
+  integrations: [
+    clerk(),
+    react()
+  ],
+  output: "server",
+  adapter: node({ mode: "standalone" }),
+});
+```
+
+### Use Clerk React components in Astro pages
+
+```astro filename="src/layouts/SiteLayout.astro"
+---
+import {
+  SignedIn,
+  SignedOut,
+  UserButton,
+  SignInButton
+} from "@clerk/astro/react";
+---
+
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+    <meta name="generator" content={Astro.generator} />
+  </head>
+  <body>
+    <header>
+      <nav>
+        <SignedOut client:load>
+         <SignInButton client:load mode="modal" />
+        </SignedOut>
+        <SignedIn client:load>
+          <UserButton client:load/>
+        </SignedIn>
+      </nav>
+    </header>
+    <article>
+      <slot />
+    </article>
+  </body>
+</html>
+```
+
+### Use Clerk React components in your React components
+
+```tsx filename="src/components/Header.tsx"
+import {
+  SignInButton,
+  SignedIn,
+  SignedOut,
+  UserButton
+} from '@clerk/astro/react'
+
+export default function Header() {
+  return (
+    <>
+      <p>My App</p>
+      <SignedOut>
+        <SignInButton />
+      </SignedOut>
+      <SignedIn>
+        <UserButton />
+      </SignedIn>
+    </>
+  )
+}
+```
+
+### Use stores in your React components
```

Comment:
Let's link to the stores section.

Reply:
Waiting on the #1259  to get merged here

Reply:
We can still add this now, if we wanted (we know what the URLs will be). I think I made a couple of suggestions that were merged, using `/docs/references/astro/overview#client-side-helpers` for the link. I used that in the suggestions since we have no overview for just the stores. This is the closet we have to an overview with a list of all of them.

Reply:
@panteliselef #1259 is in

Reply:
Addressed [here](https://github.com/clerk/clerk-docs/pull/1276)